discovery-provider/src/tasks/index_nethermind.py

```diff
@@ -66,21 +70,24 @@
 zero_address = "0x0000000000000000000000000000000000000000"
 
 
-def fetch_tx_receipt(tx_hash: HexBytes):
+class TxReceiptAndHash(TypedDict):
+    tx_receipt: TxReceipt
+    tx_hash: str
+
+
+def fetch_tx_receipt(tx_hash: HexBytes) -> TxReceiptAndHash:
     receipt = web3.eth.get_transaction_receipt(tx_hash)
-    response = {}
-    response["tx_receipt"] = receipt
-    response["tx_hash"] = tx_hash.hex()
-    return response
+    return {"tx_receipt": receipt, "tx_hash": tx_hash.hex()}
 
 
 @log_duration(logger)
-def fetch_tx_receipts(block):
-    block_transactions = block.transactions
-    block_tx_with_receipts = {}
+def fetch_tx_receipts(block: BlockData):
+    block_transactions = cast(Sequence[HexBytes], block["transactions"])
```

Comment:
casts are scary 😬 is there any way to more safely disambiguate the union type?  or is cast the "pythonic" way?

Reply:
the python way of doing this I believe is to assert to force the disambiguation 

```
assert isinstance(block["transactions"], Sequence) and all(
        isinstance(tx, HexBytes) for tx in block["transactions"]
    )
```

🤞

Reply:
^nvm type checker wasn't smart enough for this even... I think it has to be cast. :/ 

Not that I totally trust chatgpt...

<img width="664" alt="Screenshot 2023-08-07 at 12 02 26 AM" src="https://github.com/AudiusProject/audius-protocol/assets/2731362/dbbbc98d-573f-4e55-b0ca-2f28f8c41459">